packages/kbn-reporting/generate_csv/types/index.d.ts

```diff
@@ -0,0 +1,274 @@
+  }),
+  useByteOrderMarkEncoding: schema.boolean({ defaultValue: false }),
+  scroll: schema.object({
+    duration: schema.string({
+      defaultValue: '30s', // this value is passed directly to ES, so string only format is preferred
+      validate(value) {
+        if (!/^[0-9]+(d|h|m|s|ms|micros|nanos)$/.test(value)) {
+          return 'must be a duration string';
+        }
+      },
+    }),
+    size: schema.number({ defaultValue: 500 }),
+  }),
+});
+
+export type CsvConfigType = typeof CsvConfig;
+
+export interface CsvExportSettings {
+  timezone: string;
+  scroll: {
+    size: number;
+    duration: string;
+  };
+  bom: string;
+  separator: string;
+  maxSizeBytes: number | ByteSizeValue;
+  checkForFormulas: boolean;
+  escapeFormulaValues: boolean;
+  escapeValue: (value: string) => string;
+  includeFrozen: boolean;
+}
+
+export const getExportSettings = async (
+  client: IUiSettingsClient,
+  config: CsvConfigType,
+  timezone: string | undefined,
+  logger: Logger
+): Promise<CsvExportSettings> => {
+  let setTimezone: string;
+  if (timezone) {
+    setTimezone = timezone;
+  } else {
+    // timezone in settings?
+    setTimezone = await client.get(UI_SETTINGS_DATEFORMAT_TZ);
+    if (setTimezone === 'Browser') {
+      // if `Browser`, hardcode it to 'UTC' so the export has data that makes sense
+      logger.warn(
+        `Kibana Advanced Setting "dateFormat:tz" is set to "Browser". Dates will be formatted as UTC to avoid ambiguity.`
+      );
+      setTimezone = 'UTC';
+    }
+  }
+
+  // Advanced Settings that affect search export + CSV
+  const [includeFrozen, separator, quoteValues] = await Promise.all([
+    client.get(UI_SETTINGS_SEARCH_INCLUDE_FROZEN),
+    client.get(UI_SETTINGS_CSV_SEPARATOR),
+    client.get(UI_SETTINGS_CSV_QUOTE_VALUES),
+  ]);
+
+  const escapeFormulaValues = config.escapeFormulaValues;
+  const escapeValue = createEscapeValue(quoteValues, escapeFormulaValues);
+  const bom = config.useByteOrderMarkEncoding ? CSV_BOM_CHARS : '';
+
+  return {
+    timezone: setTimezone,
+    scroll: {
+      size: config.scroll.size,
+      duration: config.scroll.duration,
+    },
+    bom,
+    includeFrozen,
+    separator,
+    maxSizeBytes: config.maxSizeBytes,
+    checkForFormulas: config.checkForFormulas,
+    escapeFormulaValues,
+    escapeValue,
+  };
+};
+
+/**
+ * @deprecated
+ */
+export type BaseParamsV2 = Ensure<
+  {
+    layout?: LayoutParams;
+    objectType: string;
+    title: string;
+    browserTimezone: string; // to format dates in the user's time zone
+    version: string; // to handle any state migrations
+    locatorParams: LocatorParams[];
+  },
+  SerializableRecord
+>;
+
+/**
+ * Report job parameters that an application must return from its
+ * getSharingData function.
+ */
+
+interface BaseParamsCSV {
+  searchSource: SerializedSearchSourceFields;
+  columns?: string[];
+}
+
+export type JobParamsCSV = BaseParamsCSV & BaseParamsV2;
```

Comment:
We can just call this `JobParams` and it can be defined as:
```
interface JobParams {
  searchSource: SerializedSearchSourceFields;
  columns?: string[];
  browserTimezone?: string;
}
```

Reply:
I shoved this into the generate_csv file lmk if that seems ok thanks!